src/RA_Condition.h

```diff
@@ -39,18 +39,24 @@ inline constexpr std::array<const LPCTSTR, 13> MEMSIZE_STR
 };
 
 
-enum ComparisonType
+enum class ComparisonType : std::size_t
```

Comment:
Why specify the size of this, if not limiting it to something like 8-bits? The compiler will choose an appropriate size.

Reply:
It's not size, its underlying type. I just need it to be unsigned. Most of the at methods for the array want size_t

Reply:
I had assumed the compiler would use a smaller type for a smaller enum. It looks like it doesn't.
```
enum ComparisonType
{
    Equals,
    LessThan,
    LessThanOrEqual,
    GreaterThan,
    GreaterThanOrEqual,
    NotEqualTo,

    NumComparisonTypes
};
enum class ComparisonType2 : size_t
{
    Equals,
    LessThan,
    LessThanOrEqual,
    GreaterThan,
    GreaterThanOrEqual,
    NotEqualTo,

    NumComparisonTypes
};
static_assert(sizeof(ComparisonType) == sizeof(ComparisonType2));
```